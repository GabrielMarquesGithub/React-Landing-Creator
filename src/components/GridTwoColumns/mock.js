export default {
  title: 'Grid two columns',
  text: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Animi, illo minima. Nesciunt debitis et accusamus cupiditate, ipsam eaque, excepturi nobis deserunt inventore expedita quod beatae, voluptatibus eius quos autem natus.',
  image: 'assets/images/bgImg.svg',
  background: true,
};
