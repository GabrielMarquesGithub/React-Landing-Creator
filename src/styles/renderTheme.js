import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from './theme';

export const renderTheme = (element) => {
  return render(<ThemeProvider theme={theme}>{element}</ThemeProvider>);
};
