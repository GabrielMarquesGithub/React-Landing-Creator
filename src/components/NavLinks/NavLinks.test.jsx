import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/renderTheme';
import { NavLinks } from '.';

import mock from './mock';

describe('<NavLinks />', () => {
  it('should render links', () => {
    renderTheme(<NavLinks links={mock}>Children</NavLinks>);
    expect(screen.getAllByRole('link')).toHaveLength(mock.length);
  });

  it('should not render links', () => {
    renderTheme(<NavLinks>Children</NavLinks>);
    expect(screen.queryAllByText('Link')).toHaveLength(0);
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<NavLinks>Children</NavLinks>);
    expect(container).toMatchSnapshot();
  });
});
