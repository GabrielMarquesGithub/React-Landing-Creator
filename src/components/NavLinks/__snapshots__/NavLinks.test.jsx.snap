// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLinks /> should match snapshot 1`] = `
<div>
  <nav
    aria-label="Main menu"
    class="sc-dkzDqf liRDsc"
  />
</div>
`;
